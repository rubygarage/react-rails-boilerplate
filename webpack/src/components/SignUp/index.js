import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, intlShape } from 'react-intl';
import { Link } from 'react-router-dom';
import { Field } from 'redux-form';
import ErrorMessage from 'components/ui/ErrorMessage';
import Oauth from 'containers/Oauth';

import styles from './styles.css';

function SignUp({
  handleSubmit, submitHandler, pristine, submitting, intl: { formatMessage },
}) {
  return (
    <div className="container">
      <div className="row">

        <div className="col-md-6 offset-md-3">
          <div className={styles['register-wrap']}>
            <div className="card">
              <div className="card-header text-center">
                <h2> <FormattedMessage id="sign_up.register" /> </h2>
              </div>
              <div className="text-center">
                <Oauth type="icon-fb" provider="facebook" button />
              </div>
              <div className="card-block">
                <form onSubmit={handleSubmit(submitHandler)}>
                  <ErrorMessage name="username" />
                  <div className="input-group mb-3">
                    <Field
                      type="text"
                      component="input"
                      className="form-control"
                      name="username"
                      placeholder={formatMessage({ id: 'sign_up.username' })}
                      aria-label={formatMessage({ id: 'sign_up.username' })}
                    />
                  </div>

                  <ErrorMessage name="email" />
                  <div className="input-group mb-3">
                    <Field
                      type="text"
                      component="input"
                      className="form-control"
                      name="email"
                      placeholder={formatMessage({ id: 'sign_up.email' })}
                      aria-label={formatMessage({ id: 'sign_up.email' })}
                    />
                  </div>

                  <ErrorMessage name="password" />
                  <div className="input-group mb-3">
                    <Field
                      type="password"
                      component="input"
                      className="form-control"
                      name="password"
                      placeholder={formatMessage({ id: 'sign_up.password' })}
                      aria-label={formatMessage({ id: 'sign_up.password' })}
                    />
                  </div>

                  <ErrorMessage name="password_confirmation" />
                  <div className="input-group mb-3">
                    <Field
                      type="password"
                      component="input"
                      className="form-control"
                      name="password_confirmation"
                      placeholder={formatMessage({ id: 'sign_up.confirm_password' })}
                      aria-label={formatMessage({ id: 'sign_up.confirm_password' })}
                    />
                  </div>

                  <button className="btn btn-primary mb-3" type="submit" disabled={pristine || submitting}>
                    <FormattedMessage id="sign_up.register" />
                  </button>
                </form>

                <p>
                  <FormattedMessage id="sign_up.already_member" />
                  &nbsp;
                  <Link to="/sign_in">
                    <FormattedMessage id="sign_up.login" />
                  </Link>
                </p>

              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  );
}

SignUp.propTypes = {
  intl: intlShape,
  handleSubmit: PropTypes.func.isRequired,
  submitHandler: PropTypes.func.isRequired,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool,
};

export default SignUp;
