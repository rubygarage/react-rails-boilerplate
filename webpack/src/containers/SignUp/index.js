import React from 'react';
import { reduxForm } from 'redux-form';
import { injectIntl } from 'react-intl';

import SignUpComponent from 'components/SignUp';

import validate from './validate';
import submit from './submit';

class SignUpForm extends React.PureComponent {
  render() {
    const props = {
      ...this.props,
      submitHandler: submit,
    };

    return (
      <SignUpComponent {...props} />
    );
  }
}

export default injectIntl(reduxForm({
  form: 'signUp',
  validate,
})(SignUpForm));
