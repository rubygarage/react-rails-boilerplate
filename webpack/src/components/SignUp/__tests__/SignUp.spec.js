import React from 'react';
import { shallow } from 'enzyme';
import SignUp from 'components/SignUp';

describe('<SigniIn />', () => {
  let props = {
    handleSubmit: jest.fn(),
    submitHandler: jest.fn(),
    intl: { formatMessage: jest.fn() },
  };

  it('renders sign up form and submitting is false', () => {
    props = {
      ...props,
      submitting: false,
    };
    const component = shallow(<SignUp {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders sign up form and pristine is true', () => {
    props = {
      ...props,
      pristine: true,
    };

    const component = shallow(<SignUp {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders sign up form submitting is true', () => {
    props = {
      ...props,
      submitting: true,
    };

    const component = shallow(<SignUp {...props} />);

    expect(component).toMatchSnapshot();
  });
});
