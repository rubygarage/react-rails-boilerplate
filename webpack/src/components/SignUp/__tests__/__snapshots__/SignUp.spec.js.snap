// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SigniIn /> renders sign up form and pristine is true 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="sign_up.register"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="username"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="username"
                  type="text"
                />
              </div>
              <ErrorMessage
                name="email"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="email"
                  type="text"
                />
              </div>
              <ErrorMessage
                name="password"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <ErrorMessage
                name="password_confirmation"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password_confirmation"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={true}
                type="submit"
              >
                <FormattedMessage
                  id="sign_up.register"
                  values={Object {}}
                />
              </button>
            </form>
            <p>
              <FormattedMessage
                id="sign_up.already_member"
                values={Object {}}
              />
               
              <Link
                replace={false}
                to="/sign_in"
              >
                <FormattedMessage
                  id="sign_up.login"
                  values={Object {}}
                />
              </Link>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SigniIn /> renders sign up form and submitting is false 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="sign_up.register"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="username"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="username"
                  type="text"
                />
              </div>
              <ErrorMessage
                name="email"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="email"
                  type="text"
                />
              </div>
              <ErrorMessage
                name="password"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <ErrorMessage
                name="password_confirmation"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password_confirmation"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={false}
                type="submit"
              >
                <FormattedMessage
                  id="sign_up.register"
                  values={Object {}}
                />
              </button>
            </form>
            <p>
              <FormattedMessage
                id="sign_up.already_member"
                values={Object {}}
              />
               
              <Link
                replace={false}
                to="/sign_in"
              >
                <FormattedMessage
                  id="sign_up.login"
                  values={Object {}}
                />
              </Link>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SigniIn /> renders sign up form submitting is true 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="sign_up.register"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="username"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="username"
                  type="text"
                />
              </div>
              <ErrorMessage
                name="email"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="email"
                  type="text"
                />
              </div>
              <ErrorMessage
                name="password"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <ErrorMessage
                name="password_confirmation"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password_confirmation"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={true}
                type="submit"
              >
                <FormattedMessage
                  id="sign_up.register"
                  values={Object {}}
                />
              </button>
            </form>
            <p>
              <FormattedMessage
                id="sign_up.already_member"
                values={Object {}}
              />
               
              <Link
                replace={false}
                to="/sign_in"
              >
                <FormattedMessage
                  id="sign_up.login"
                  values={Object {}}
                />
              </Link>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
